daml-assistant/src/DAML/Assistant/Install.hs

```diff
@@ -173,10 +184,44 @@ activateDaml env@InstallEnv{..} targetPath = do
                      ]
             else createSymbolicLink damlBinarySourcePath damlBinaryTargetPath
 
-    unlessQuiet env $ do -- Ask user to add .daml/bin to PATH if it is absent.
-        searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
-        when (damlBinaryTargetDir `notElem` searchPaths) $ do
-            output ("Please add " <> damlBinaryTargetDir <> " to your PATH.")
+    updatePath env damlBinaryTargetDir
+
+updatePath :: InstallEnv -> FilePath -> IO ()
+#ifdef mingw32_HOST_OS
+updatePath env@InstallEnv{..} targetPath = do
+    -- Updating PATH on Windows is annoying so we do it automatically.
+    bracket (regOpenKeyEx hKEY_CURRENT_USER "Environment" kEY_ALL_ACCESS) regCloseKey $ \envKey -> do
+        path <- regQueryValueEx envKey "Path"
+        let newPath = targetPath <> ";" <> remove targetPath path
+        regSetStringValue envKey "Path" newPath
+    unlessQuiet env $
+        output (targetPath <> " has been added to your PATH. Reboot for the changes to take effect.")
+    where
+        -- Given a comma separated string, remove any occurences of a given string.
+        remove :: String -> String -> String
+        remove x = intercalate ";" . filter (/= x) . split (== ';')
+
+-- | Win32 does expose function of the same name but it requires you to know the
+-- size of the value in advance which makes it quite annoying to use.
+regQueryValueEx :: HKEY -> String -> IO String
+regQueryValueEx key valueName =
+    withForeignPtr key $ \p_key ->
+    withTString valueName $ \c_valueName ->
+    alloca $ \p_valueLen -> do
+        failUnlessSuccess "regQueryDefaultValue" $
+            c_RegQueryValueEx p_key c_valueName nullPtr nullPtr nullPtr p_valueLen
+        valueLen <- peek p_valueLen
+        allocaArray0 (fromIntegral valueLen) $ \ c_value -> do
+            failUnlessSuccess "regQueryDefaultValue" $
+                c_RegQueryValueEx p_key c_valueName nullPtr nullPtr c_value p_valueLen
+            peekTString (castPtr c_value)
+#else
+updatePath env@InstallEnv{..} targetPath = unlessQuiet env $ do
+    -- Ask user to add .daml/bin to PATH if it is absent.
+    searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
+    when (targetPath `notElem` searchPaths) $ do
```

Comment:
Could we have this check on both branches? If the user has rejigged their path to put DAML at the end for some reason, its impolite to move it to the start.

Reply:
:+1: